### DISCUSSION TITLE
UrlFetch from privat Git repo

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#21160] [Q&A]
i have a url that works in browser and load directly the raw file (berry script).
`http://<user>:<acces-tocken>@<server-ip>/<group>/<project_name>/-/raw/<branch>/<folder>/controll.be`
or
`http://<acces-tocken>@<server-ip>/<group>/<project_name>/-/raw/<branch>/<folder>/controll.be`

but doing this from Tasmota console with `UrlFetch` i got a file named `controll.be` but with html content of git login.

**Comment by barbudor** [comment]
Sorry I didn't saw that you were already using a token

**Comment by sfromis** [comment]
If you need to amend the way it works, like adding headers, it is a pretty simple Berry function you can copy and play with.
https://github.com/arendst/Tasmota/blob/development/lib/libesp32/berry_tasmota/src/embedded/tasmota_class.be#L796-L816

AFAIK, the browser is "cheating" a bit, to be helpful, by expanding your URL into following the basic authentication protocol, differently from when Berry just uses the URL directly.

**Comment by ChrisVonFrosta** [comment]
Thank you. I gave it a short try but without success. Actually, I found a workaround to get the 'large' Berry files uploaded via Windows prompt most of the time.
